perception/autoware_image_projection_based_fusion/include/autoware/image_projection_based_fusion/fusion_node.hpp

```diff
@@ -52,14 +52,34 @@ using autoware_perception_msgs::msg::DetectedObject;
 using autoware_perception_msgs::msg::DetectedObjects;
 using sensor_msgs::msg::CameraInfo;
 using sensor_msgs::msg::Image;
-using sensor_msgs::msg::PointCloud2;
-using tier4_perception_msgs::msg::DetectedObjectsWithFeature;
+using PointCloudMsgType = sensor_msgs::msg::PointCloud2;
+using RoiMsgType = tier4_perception_msgs::msg::DetectedObjectsWithFeature;
+using ClusterMsgType = tier4_perception_msgs::msg::DetectedObjectsWithFeature;
+using ClusterObjType = tier4_perception_msgs::msg::DetectedObjectWithFeature;
 using tier4_perception_msgs::msg::DetectedObjectWithFeature;
 using PointCloud = pcl::PointCloud<pcl::PointXYZ>;
 using autoware::image_projection_based_fusion::CameraProjection;
 using autoware_perception_msgs::msg::ObjectClassification;
 
-template <class TargetMsg3D, class ObjType, class Msg2D>
+template <class Msg2D>
+struct Det2dManager
+{
+  // camera index
+  std::size_t id = 0;
+  // camera projection
+  std::unique_ptr<CameraProjection> camera_projector_ptr;
+  bool project_to_unrectified_image = false;
+  bool approximate_camera_projection = false;
+  // process flags
+  bool is_fused = false;
+  // timing
+  double input_offset_ms = 0.0;
+  // cache
+  std::map<int64_t, typename Msg2D::ConstSharedPtr> cached_det2d_msgs;
+  std::unique_ptr<std::mutex> mtx_ptr;
+};
```

Comment:
IMHO it's better to explicitly separate mutable and immutable variables. For instance `id`, `project_to_unrectified_image`, or `approximate_camera_projection` is immutable while `cached_det2d_msgs` or `is_fused` is mutable, right? 

(Below is just a random implementation proposal, but maybe there is a better way to achieve this)
```suggestion
template <class Msg2D>
struct Det2dManager
{
  // camera index
  const std::size_t id;
  // camera projection
  const std::unique_ptr<CameraProjection> camera_projector_ptr;
  const bool project_to_unrectified_image;
  const bool approximate_camera_projection;
  // process flags
  bool is_fused = false;
  // timing
  const double input_offset_ms;
  // cache
  std::map<int64_t, typename Msg2D::ConstSharedPtr> cached_det2d_msgs;
  std::unique_ptr<std::mutex> mtx_ptr;
  
  Det2dManager(std::size_t id_, ...): id(id_), ... {}
};
```

Reply:
It is a `struct` and replaces vector of variables.
In terms of readability, the current `struct` approach is appropriate.
I expect developers use it properly.